src/epd2in13_v2/command.rs

```diff
@@ -0,0 +1,121 @@
+//! SPI Commands for the Waveshare 2.13" v2
+
+use crate::traits;
+
+/// EPD2in13 v2
+///
+/// Should rarely (never?) be needed directly.
+///
+/// For more infos about the addresses and what they are doing look into the pdfs
+#[allow(dead_code)]
+#[allow(non_camel_case_types)]
+#[derive(Copy, Clone)]
+pub(crate) enum Command {
+    /// Driver Output control
+    ///     3 Databytes:
+    ///     A[7:0]
+    ///     0.. A[8]
+    ///     0.. B[2:0]
+    ///     Default: Set A[8:0] = 0x127 and B[2:0] = 0x0
+    DRIVER_OUTPUT_CONTROL = 0x01,
+
+    GATE_DRIVING_VOLTAGE_CTRL = 0x03,
+    SOURCE_DRIVING_VOLTAGE_CTRL = 0x04,
+
+    /// Booster Soft start control
+    ///     3 Databytes:
+    ///     1.. A[6:0]
+    ///     1.. B[6:0]
+    ///     1.. C[6:0]
+    ///     Default: A[7:0] = 0xCF, B[7:0] = 0xCE, C[7:0] = 0x8D
+    BOOSTER_SOFT_START_CONTROL = 0x0C,
+    GATE_SCAN_START_POSITION = 0x0F,
+    //TODO: useful?
+    // GATE_SCAN_START_POSITION = 0x0F,
+    /// Deep Sleep Mode Control
+    ///     1 Databyte:
+    ///     0.. A[0]
+    ///     Values:
+    ///         A[0] = 0: Normal Mode (POR)
+    ///         A[0] = 1: Enter Deep Sleep Mode
+    DEEP_SLEEP_MODE = 0x10,
+    // /// Data Entry mode setting
+    DATA_ENTRY_MODE_SETTING = 0x11,
+
+    SW_RESET = 0x12,
+
+    TEMPERATURE_SENSOR_CONTROL_WRITE = 0x1A,
+    TEMPERATURE_SENSOR_CONTROL_READ = 0x1B,
+
+    MASTER_ACTIVATION = 0x20,
+
+    DISPLAY_UPDATE_CONTROL_1 = 0x21,
+
+    DISPLAY_UPDATE_CONTROL_2 = 0x22,
+
+    WRITE_RAM = 0x24,
+    READ_RAM = 0x27,
+
+    VCOM_SENSE = 0x28,
+
+    WRITE_VCOM_REGISTER = 0x2C,
+
+    WRITE_LUT_REGISTER = 0x32,
+
+    WRITE_REGISTER_TO_DISPLAY_OPT = 0x37,
+    WRITE_REGISTER_FOR_USER_ID = 0x38,
+
+    SET_DUMMY_LINE_PERIOD = 0x3A,
+
+    SET_GATE_LINE_WIDTH = 0x3B,
+
+    BORDER_WAVEFORM_CONTROL = 0x3C,
+
+    SET_RAM_X_ADDRESS_START_END_POSITION = 0x44,
+
+    SET_RAM_Y_ADDRESS_START_END_POSITION = 0x45,
+
+    SET_RAM_X_ADDRESS_COUNTER = 0x4E,
+
+    SET_RAM_Y_ADDRESS_COUNTER = 0x4F,
+
+    SET_ANALOG_BLOCK_CONTROL = 0x74,
+
+    SET_DIGITAL_BLOCK_CONTROL = 0x7E,
+
+    NOP = 0x7F,
+}
+
+pub(crate) enum DataEntryModeIncr {
+    X_DECR_Y_DECR = 0x0,
+    X_INCR_Y_DECR = 0x1,
+    X_DECR_Y_INCR = 0x2,
+    X_INCR_Y_INCR = 0x3
+}
```

Comment:
what does dataentrymode do?
Can you only change some partial parts of the epd ram/gates and also fill it in a backwards order for the 2in13v2?

Reply:
When you send bytes with `WRITE_RAM` it controls how the target pointer is modified after each bytes. The pointer can increase in both direction (seems the default in your code), but you can have x increased and y decreased (that's what the sample code from waveshare is doing).